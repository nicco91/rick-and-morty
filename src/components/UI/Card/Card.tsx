import classNames from 'classnames';
import React, { FC } from 'react';
import styled from 'styled-components';

type Props = {
  className?: string;
  image?: string;
  title: string;
  subTitle: React.ReactNode;
  extra?: React.ReactNode;
};

const CardWrapper = styled.div`
  border-radius: ${(props) => props.theme.borderRadius};
  box-shadow: ${(props) => props.theme.boxShadow};
  background-color: white;
  display: flex;
  flex-direction: column;
  position: relative;
  overflow: hidden;
  height: 100%;
`;

const CardImage = styled.div<{ image: string }>`
  background-image: ${({ image }) => `url('${image}')`};
  background-position: center center;
  background-size: cover;
  min-height: 300px;
`;

const CardBody = styled.div`
  padding: 16px;
  box-sizing: border-box;
`;

const CardHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 8px;
`;

const CardTitle = styled.h3`
  margin: 0;
  color: ${(props) => props.theme.colors.textPrimary};
  font-size: 24px;
`;

const CardSubtitle = styled.p`
  margin-top: 0;
  color: ${(props) => props.theme.colors.textSecondary};
  font-size: 18px;
`;

const Card: FC<Props> = ({ className, image, title, subTitle, extra, children }) => {
  return (
    <CardWrapper className={classNames('Card', className)}>
      {image && <CardImage image={image} />}
      <CardBody>
        <CardHeader>
          <CardTitle>{title}</CardTitle>
          <div>{extra}</div>
        </CardHeader>
        <CardSubtitle>{subTitle}</CardSubtitle>
        <div>{children}</div>
      </CardBody>
    </CardWrapper>
  );
};

export default Card;
