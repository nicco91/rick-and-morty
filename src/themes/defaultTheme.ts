import { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#0066cc',
    success: '#00cc66',
    error: '#cc0000',
    textPrimary: '#333',
    textSecondary: '#999',
  },
  breakpoints: {
    mobile: 320,
    mobileLandscape: 480,
    tablet: 768,
    tabletLandscape: 1024,
    desktop: 1200,
    desktopLarge: 1500,
    desktopWide: 1920,
  },
};
