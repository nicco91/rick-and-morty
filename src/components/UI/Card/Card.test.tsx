import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'themes';
import Card from './Card';

it('render a card', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <Card title="Title" subTitle="Sub title">
          I am a card!
        </Card>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
