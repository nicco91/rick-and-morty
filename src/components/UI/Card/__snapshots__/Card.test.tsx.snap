// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a card 1`] = `
.c0 {
  border-radius: 16px;
  box-shadow: rgba(0,0,0,0.1) 0px 10px 15px -3px,rgba(0,0,0,0.05) 0px 4px 6px -2px;
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  overflow: hidden;
  height: 100%;
}

.c1 {
  padding: 16px;
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.c3 {
  margin: 0;
  color: #333;
  font-size: 24px;
}

.c4 {
  margin-top: 0;
  color: #999;
  font-size: 18px;
}

<div
  className="c0 Card"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h3
        className="c3"
      >
        Title
      </h3>
      <div />
    </div>
    <p
      className="c4"
    >
      Sub title
    </p>
    <div>
      I am a card!
    </div>
  </div>
</div>
`;
